lib/identity_cache.rb

```diff
@@ -40,8 +40,18 @@ class << self
     mattr_accessor :cache_namespace
     self.cache_namespace = "IDC:#{CACHE_VERSION}:".freeze
 
+    version = Gem::Version.new(IdentityCache::VERSION)
+
+    # fetch_#{association} for a cache_has_many association returns a relation
+    # when fetch_returns_relation is set to true and an array when set to false.
     mattr_accessor :fetch_returns_relation
-    self.fetch_returns_relation = true
+    self.fetch_returns_relation = version >= Gem::Version.new("0.4")
```

Comment:
self.fetch_returns_relation should return true for versions < 0.4 to keep it consistent with what we had before ... maybe try `self.fetch_returns_relation = version < Gem::Version.new("0.4")` ?